Add changelog for v0.5.0.

ehpor opened the pull request:
Add changelog for v0.5.0.

syhaffert commented on doc/changelog.rst:
```diff
@@ -1,6 +1,35 @@
+* Improved support for animations (`@ehpor <https://github.com/ehpor>`__)
+* Added focal-length parameter to Lyot coronagraph (`@syhaffert <https://github.com/syhaffert>`__)
+* Added a telescope pupil introductory tutorial (`@ehpor <https://github.com/ehpor>`__)
+* Allow single-mode fibers to be put at arbitrary positions (`@syhaffert <https://github.com/syhaffert>`__)
```
step-index fibers (can have any amount of modes).

ehpor replied:
Fixed.

syhaffert commented on doc/changelog.rst:
```diff
@@ -1,6 +1,35 @@
+* Autodetect FFTs when doing general Fourier transforms (`@ehpor <https://github.com/ehpor>`__)
+* Added file IO for Field, Grid and ModeBasis (`@ehpor <https://github.com/ehpor>`__)
+* Allow the use of 32bit floating point Wavefronts (`@ehpor <https://github.com/ehpor>`__)
+* Added the LUVOIR-B, GMT, TMT, Hale, HabEx, Hubble, JWST and VLTI telescope apertures (`@ivalaginja <https://github.com/ivalaginja>`__, `@syhaffert <https://github.com/syhaffert>`__ & `@ehpor <https://github.com/ehpor>`__)
```
let's leave the VLTI for the next release. I think the ELT was also added in this release.

ehpor replied:
Agreed about VLTI. I might have missed ELT. Let me check.

ehpor replied:
Yep, I missed the ELT. I don't know how, since it's extremely large. Fixed.

syhaffert replied:
Yeah it was part of my major PR that included multiple things.

ehpor replied:
Makes sense. I mostly looked at the names of merged PRs.

ehpor replied:
One more reason to automate the writing of release notes, tbh.

ehpor commented:
Rebased and added release date.

Ran all tests (also long-running) locally and everything passed. Built documentation without (relevant) errors/warnings.

syhaffert reviewed (approved):
Time for merging.
